src/js/game/components/item_acceptor.js

```diff
@@ -1,27 +1,48 @@
 import { enumDirection, enumInvertedDirections, Vector } from "../../core/vector";
-import { types } from "../../savegame/serialization";
 import { BaseItem } from "../base_item";
 import { Component } from "../component";
+import { Entity } from "../entity";
+import { GameRoot } from "../root";
 
-/** @typedef {{
+/**
+ * @typedef {{
  * pos: Vector,
- * directions: enumDirection[],
+ * direction: enumDirection,
  * filter?: ItemType
- * }} ItemAcceptorSlot */
-
-/**
+ * }} ItemAcceptorSlot
+ *
  * Contains information about a slot plus its location
  * @typedef {{
- *  slot: ItemAcceptorSlot,
- *  index: number,
- *  acceptedDirection: enumDirection
- * }} ItemAcceptorLocatedSlot */
-
-/** @typedef {{
+ * slot: ItemAcceptorSlot,
+ * index: number,
+ * acceptedDirection: enumDirection
+ * }} ItemAcceptorLocatedSlot
+ *
+ * @typedef {{
  * pos: Vector,
- * directions: enumDirection[],
+ * direction: enumDirection,
  * filter?: ItemType
- * }} ItemAcceptorSlotConfig */
+ * }} ItemAcceptorSlotConfig
+ *
+ * @typedef {Map<number, {

```

Comment:
Can you change it to `Object<number, ...>` and also change the code to use `{}` instead of `new Map`?

Reply:
using arrays in the end